### PR TITLE
Fix nginx restarting issue by waiting for the nginx configuration fil…

### LINKED ISSUE #37: Nginx container starts after second node deployment during artifactory-ha helm chart deployment
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

**Version of Helm and Kubernetes**:
0.4.0

**Which chart**:
artifactory-ha

**What happened**:
After deploying the chart, the Nginx container starts up and the container's lifecycle postStart event handler is kicked off but then the container is killed off and a new one is created.  Please note the following from the log below:

"Killing container with id docker://nginx:FailedPostStartHook"

  Normal   Pulling                6m    kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  pulling image "docker.bintray.io/jfrog/nginx-artifactory-pro:6.2.0"
  Normal   Pulled                 4m    kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Successfully pulled image "docker.bintray.io/jfrog/nginx-artifactory-pro:6.2.0"
  Warning  FailedPostStartHook    4m    kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Exec lifecycle hook ([/bin/sh -c if [ -f /tmp/replicator-nginx.conf ]; then cp -fv /tmp/replicator-nginx.conf /etc/nginx/conf.d/replicator-nginx.conf; fi; if [ -f /tmp/ssl/*.crt ]; then rm -rf /var/opt/jfrog/nginx/ssl/example.*; cp -fv /tmp/ssl/* /var/opt/jfrog/nginx/ssl; fi; until [ -f /etc/nginx/conf.d/artifactory.conf ]; do sleep 1; done; if ! grep -q 'upstream' /etc/nginx/conf.d/artifactory.conf; then sed -i -e 's,proxy_pass .*,proxy_pass     http://art-artifactory-ha:8081/artifactory/;,g' \
    -e 's,server_name .*,server_name ~(?<repo>.+)\\.art-artifactory-ha art-artifactory-ha;,g' \
    /etc/nginx/conf.d/artifactory.conf;
fi; if ! grep -q 'proxy_http_version' /etc/nginx/conf.d/artifactory.conf; then sed -i 's,\(proxy_next_upstream .*\),proxy_http_version  1.1;\n    \1,g' /etc/nginx/conf.d/artifactory.conf; fi; sleep 5; nginx -s reload; touch /var/log/nginx/conf.done
]) for Container "nginx" in Pod "art-nginx-7f94bb7698-lhb5h_osh-infra(bc9c2801-bb5a-11e8-9ab6-fa163ec90eef)" failed - error: command '/bin/sh -c if [ -f /tmp/replicator-nginx.conf ]; then cp -fv /tmp/replicator-nginx.conf /etc/nginx/conf.d/replicator-nginx.conf; fi; if [ -f /tmp/ssl/*.crt ]; then rm -rf /var/opt/jfrog/nginx/ssl/example.*; cp -fv /tmp/ssl/* /var/opt/jfrog/nginx/ssl; fi; until [ -f /etc/nginx/conf.d/artifactory.conf ]; do sleep 1; done; if ! grep -q 'upstream' /etc/nginx/conf.d/artifactory.conf; then sed -i -e 's,proxy_pass .*,proxy_pass     http://art-artifactory-ha:8081/artifactory/;,g' \
    -e 's,server_name .*,server_name ~(?<repo>.+)\\.art-artifactory-ha art-artifactory-ha;,g' \
    /etc/nginx/conf.d/artifactory.conf;
fi; if ! grep -q 'proxy_http_version' /etc/nginx/conf.d/artifactory.conf; then sed -i 's,\(proxy_next_upstream .*\),proxy_http_version  1.1;\n    \1,g' /etc/nginx/conf.d/artifactory.conf; fi; sleep 5; nginx -s reload; touch /var/log/nginx/conf.done
' exited with 137: , message: ""
  **Normal  Created  4m (x2 over 4m)  kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Created container
  Normal  Killing  4m               kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Killing container with id docker://nginx:FailedPostStartHook
  Normal  Pulled   4m               kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Container image "docker.bintray.io/jfrog/nginx-artifactory-pro:6.2.0" already present on machine
  Normal  Started  4m (x2 over 4m)  kubelet, artifactory-ha-1-openstack-helm-server-u4publsxw2wc  Started container**

**What you expected to happen**:
 The Nginx container and the postStart event should run and be successful and a restart of the container shouldn't be happening.  Otherwise, the Nginx configMap from nginx-replicator-conf.yaml may not be used.

**How to reproduce it** (as minimally and precisely as possible):
The issue is intermittent and looks to be some sort of race condition. When it does happen, the Nginx container starts up after the second Artifactory node (artifactory-ha-artifactory-ha-member-0) and before the third (artifactory-ha-artifactory-ha-member-1).  Then a restart happens on the Nginx container which is causing a longer than expected deployment.

**kubectl get pods** (ran after second node deployment)
NAME READY STATUS RESTARTS AGE
artifactory-ha-artifactory-ha-member-0 1/1 Running 0 4m
artifactory-ha-artifactory-ha-member-1 0/1 Running 0 1m
artifactory-ha-artifactory-ha-primary-0 1/1 Running 0 4m
artifactory-ha-nginx-57df8c7d66-vdjcv 1/1 Running 1 4m
artifactory-ha-postgresql-9d4dc6bc7-rm8rc 1/1 Running 0 4m

**kubectl describe pods artifactory-ha-nginx-57df8c7d66-vdjcv**
Normal Killing 2m kubelet, gke-marcb-cluster-default-pool-2ab0ef6f-559k Killing container with id docker://nginx:FailedPostStartHook
Normal Created 2m (x2 over 2m) kubelet, gke-marcb-cluster-default-pool-2ab0ef6f-559k Created container
Normal Started 2m (x2 over 2m) kubelet, gke-marcb-cluster-default-pool-2ab0ef6f-559k Started container
Normal Pulled 2m kubelet, gke-marcb-cluster-default-pool-2ab0ef6f-559k Container image "docker.bintray.io/jfrog/nginx-artifactory-pro:6.3.2" already present on machine

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by danielezer**
…e to be there before starting to use it

#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
Fix the nginx restarting issue by waiting for the nginx configuration file to appear before using it

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #37 

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/artifactory-ha/templates/nginx-deployment.yaml [hunk 1]
```diff
@@ -72,6 +72,7 @@ spec:
                 {{- if .Values.nginx.customArtifactoryConfigMap }}
                 cp -Lrf /tmp/conf.d/artifactory*.conf /etc/nginx/conf.d/artifactory.conf;
                 {{- else }}
+                until [ -f /etc/nginx/conf.d/artifactory.conf ]; do echo ; done;
```
Comment (eldada) [inline stable/artifactory-ha/templates/nginx-deployment.yaml]: Please add a `sleep 1` in the loop so CPU will not spike.

File: stable/artifactory/templates/nginx-deployment.yaml [hunk 2]
```diff
@@ -72,6 +72,7 @@ spec:
                 {{- if .Values.nginx.customArtifactoryConfigMap }}
                 cp -Lrf /tmp/conf.d/artifactory*.conf /etc/nginx/conf.d/artifactory.conf;
                 {{- else }}
+                until [ -f /etc/nginx/conf.d/artifactory.conf ]; do echo; done;
```
Comment (eldada) [inline stable/artifactory/templates/nginx-deployment.yaml]: Please add a `sleep 1` in the loop so CPU will not spike.

Comment (danielezer) [reply stable/artifactory/templates/nginx-deployment.yaml]: Donw

File: stable/artifactory-ha/templates/nginx-deployment.yaml (hunk 1 above)
Comment (danielezer) [reply stable/artifactory-ha/templates/nginx-deployment.yaml]: Done

**Comment by danielezer** [comment]
@eldada please approve as well :)

**Pull request merged**